File: petclinic-data/src/main/java/xyz/haff/petclinic/models/Vet.java
package xyz.haff.petclinic.models;

import lombok.*;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.List;

@Entity
@Getter
@Setter
@ToString(callSuper = true, onlyExplicitlyIncluded = true)
@NoArgsConstructor
@SuperBuilder
public class Vet extends Person {
    @Enumerated(EnumType.STRING)
    @Column(name = "specialty")
    @ToString.Include
    @NotNull
    private Specialty specialty;

    @OneToMany(mappedBy = "vet")
    private List<Visit> visits;
}

File: petclinic-data/src/main/kotlin/xyz/haff/petclinic/models/Vet.kt
package xyz.haff.petclinic.models

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Version
import java.util.*

data class Vet @JvmOverloads constructor(
    val baseEntity: BaseEntity = BaseEntity(),
    val personalData: PersonalData
): Entity by baseEntity